api/src/services/user-service.ts

```diff
@@ -92,6 +92,22 @@ export class UserService {
         }
     }
 
+    public static async getUserPasswordHashByUserID(options: {
+        userID: string;
+    }): Promise<string> {
+        const userRepository = this.getUserRepository();
+        try {
+            let user = await userRepository
+                .createQueryBuilder("user")
```

Comment:
Because we need access to the hashed password. as far as i understand the code, we don't expose the password normally. We only ever used it in the auth part of the code and there we use a query builder to explicitally access the password. Thats why I implemented it the same way here.